tests/common/platform/interface_utils.py

```diff
@@ -121,3 +124,26 @@ def get_port_map(dut, asic_index=None):
         port_mapping[k] = [v]
 
     return port_mapping
+
+def get_physical_port_indices(duthost):
+    """Returns list of physical port numbers of the DUT"""
+    physical_port_indices = set()
+
+    intf_facts = duthost.interface_facts()['ansible_facts']['ansible_interface_facts']
+    phy_port = re.compile(r'^Ethernet\d+$')
+    phy_intfs = [k for k in intf_facts.keys() if re.match(phy_port, k)]
+    phy_intfs = natsorted(phy_intfs)
+    logging.info("physical interfaces = {}".format(phy_intfs))
+
+    for asic_index in duthost.get_frontend_asic_ids():
+	# Get interfaces of this asic
+	interface_list = get_port_map(duthost, asic_index)
+	interfaces_per_asic = {k:v for k, v in interface_list.items() if k in phy_intfs}
+	#logging.info("ASIC index={} interfaces = {}".format(asic_index, interfaces_per_asic))
+	asichost = duthost.asic_instance(asic_index)
+	for intf in interfaces_per_asic:
+	    cmd = 'redis-cli -n 4 hget "PORT|{}" index'.format(intf)
+	    docker_cmd = asichost.get_docker_cmd(cmd, "database")
```

Comment:
You could call sonic-db-cli from base image, no need to get into the database docker and call redis-cli from there. Is there a special consideration you did so?

Reply:
@yxieca no special consideration, its just that I found asichost.get_docker_cmd() more convenient to use here. do you see any issue?

Reply:
sonic-db-cli is more generic and less likely to change across versions. But this is not a blocker, we can deal with that when we cross that bridge.